src/tir/usmp/analysis/extract_buffer_info.cc

```diff
@@ -491,6 +559,55 @@ BufferInfoAnalysis BufferInfoExtractor::operator()(const PrimFunc& main_func) {
       open_set.erase(le_event.buffer_info);
     }
   }
+
+  // All items in RO pool should have conflicts with each other in this RO pool
+  // as they will be placed in RO segment and pre-initialized
+  // Array<BufferInfo> buffer_info_arr =
+  //    ConvertToArrayOfBufferInfo(this->buffer_info_map_);
+
+  // split buffers to vars (RW) and constants (RO)
+  Array<BufferInfo> buffer_info_vars;
+  Array<BufferInfo> buffer_info_cons;
+  for (const auto& kv : this->buffer_info_map_) {
+    const auto& buf = kv.first;
+    for (const auto& pool : buf->pool_candidates) {
```

Comment:
Can we not just check kv.second for the node type instead to determine this ? 
Which would simplify the code here.

Reply:
done